native/src/components/PrimaryTextButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactElement } from 'react'
+import { Button } from 'react-native-elements'
+import { IconNode } from 'react-native-elements/dist/icons/Icon'
+import styled, { useTheme } from 'styled-components/native'
+
+const Container = styled.View<{ $padding: boolean }>`
+  padding: ${props => (props.$padding ? '16px' : '0')};
+`
+
+type PrimaryTextButtonProps = {
+  onPress: () => void
+  text: string
+  disabled?: boolean
+  Icon?: IconNode
```

Comment:
i'm really not a big fan to be bound to icons from a specific library for a base component. What if want to have a specific svg icon here in future?